kong/plugins/moesif/regex_config_helpers.lua

```diff
@@ -0,0 +1,180 @@
+local _M = {}
+
+-- Function to perform the regex matching with event value and condition value
+-- @param  `event_value`     Value associated with event (request)
+-- @param  `condition_value` Value associated with the regex config condition
+-- @param  `regex_matched`   Boolean flag with regex matched from previous regex config condition
+-- @return `regex_matched`   Boolean flag to determine if the regex match was successful 
+local function regex_match (event_value, condition_value, regex_matched)
+    -- Perform regex match between event value and regex config condition value
+    if string.match(event_value, condition_value) then 
```

Comment:
this is not efficient right. if regex_matched is false, then there is no point in doing string.match again ..because its a 'and'. If previous is false, then end result is false.

Reply:
may be pass initial value as true..that way dont need to do this nil checking..
this fcn can be simplified